pkg/scheduler/actions/allocate/allocate.go

```diff
@@ -125,34 +127,47 @@ func (alloc *allocateAction) Execute(ssn *framework.Session) {
 					glog.V(3).Infof("Predicates failed for task <%s/%s> on node <%s>: %v",
 						task.Namespace, task.Name, node.Name, err)
 					continue
+				} else {
+					predicateNodes = append(predicateNodes, node)
+				}
+			}
+			for _, node := range predicateNodes {
+				score, err := ssn.PriorityFn(task, node, predicateNodes)
```

Comment:
why we need `predicateNodes`? it's better to use all nodes, there're already some enhancement in upstream.

Reply:
We need to run priority on predicated nodes right? If we run priority in every nodes and a node which does not satify predicates gets high score, then it will try to Allocate on that node right? So we should run priority on nodes which has already satisfied all the predicates defined right?

Reply:
> We need to run priority on predicated nodes right?

Yes, but why pass all predicateNodes to priority fn?

Reply:
InterPodAffinityPriority function expects set of nodes as input, so we had to send all the predicated nodes to PriorityFn
```suggestion
CalculateInterPodAffinityPriority(pod *v1.Pod, nodeNameToInfo map[string]*schedulercache.NodeInfo, nodes []*v1.Node)
```

Reply:
`InterPodAffinityPriority` should use all nodes instead of predicated nodes for topology, as some pods are in topology but not in predicated nodes. It should be handled in upstream.